process/sync/storageBootstrap/baseStorageBootstrapper.go

```diff
@@ -0,0 +1,315 @@
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/block/bootstrapStorage"
+	"github.com/ElrondNetwork/elrond-go/process/sync"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+var log = logger.GetOrCreate("process/sync")
+
+// ArgsStorageBootstrapper is structure used to create a new storage bootstrapper
+type ArgsStorageBootstrapper struct {
+	BootStorer          process.BootStorer
+	ForkDetector        process.ForkDetector
+	BlockProcessor      process.BlockProcessor
+	ChainHandler        data.ChainHandler
+	Marshalizer         marshal.Marshalizer
+	Store               dataRetriever.StorageService
+	Uint64Converter     typeConverters.Uint64ByteSliceConverter
+	BootstrapRoundIndex uint64
+	ShardCoordinator    sharding.Coordinator
+	ResolversFinder     dataRetriever.ResolversFinder
+}
+
+type storageBootstrapper struct {
+	bootStorer       process.BootStorer
+	forkDetector     process.ForkDetector
+	blkExecutor      process.BlockProcessor
+	blkc             data.ChainHandler
+	marshalizer      marshal.Marshalizer
+	store            dataRetriever.StorageService
+	uint64Converter  typeConverters.Uint64ByteSliceConverter
+	shardCoordinator sharding.Coordinator
+
+	bootstrapRoundIndex  uint64
+	bootstrapper         storageBootstrapperHandler
+	headerNonceHashStore storage.Storer
+}
+
+func (st *storageBootstrapper) loadBlocks() error {
+	var err error
+	var storageHeaderInfo bootstrapStorage.BootstrapData
+
+	round := st.bootStorer.GetHighestRound()
+	storageHeadersInfo := make([]bootstrapStorage.BootstrapData, 0)
+
+	log.Debug("Load blocks started...")
+
+	for {
+		storageHeaderInfo, err = st.bootStorer.Get(round)
+		if err != nil {
+			break
+		}
+		storageHeadersInfo = append(storageHeadersInfo, storageHeaderInfo)
+
+		if uint64(round) > st.bootstrapRoundIndex {
+			round = storageHeaderInfo.LastRound
+			continue
+		}
+
+		err = st.applyHeaderInfo(storageHeaderInfo)
+		if err != nil {
+			round = storageHeaderInfo.LastRound
+			continue
+		}
+
+		bootInfos, err := st.getBootInfos(storageHeaderInfo)
+		if err != nil {
+			round = storageHeaderInfo.LastRound
+			continue
+		}
+
+		err = st.applyBootInfos(bootInfos)
+		if err != nil {
+			round = storageHeaderInfo.LastRound
+			continue
+		}
+
+		break
+	}
+
+	if err != nil {
+		_ = st.bootStorer.SaveLastRound(0)
+		return process.ErrNotEnoughValidBlocksInStorage
+	}
+
+	var processedMiniBlocks map[string]map[string]struct{}
+	errNotCritical := st.marshalizer.Unmarshal(&processedMiniBlocks, storageHeaderInfo.ProcessedMiniBlocksBytes)
+	if errNotCritical != nil && storageHeaderInfo.ProcessedMiniBlocksBytes != nil {
+		log.Debug("cannot unmarshal base storage bootstrapper ",
+			"error", errNotCritical.Error())
+	}
+
+	log.Debug("processed mini blocks applied")
```

Comment:
The following code block could be extracted in a separate function

Reply:
done.